src/main/java/org/candlepin/subscriptions/metering/MeteringEventFactory.java

```diff
@@ -98,6 +100,24 @@ public static Event createMetricEvent(
     return event;
   }
 
+  /**
+   * Creates an event with type "cleanup" which is a special type that triggers the deletion of
+   * stale events.
+   *
+   * @param orgId the organization id.
+   * @param eventType the event type.
+   * @param cutOff the time when events are considered stale events.
+   * @return a populated Event instance.
+   */
+  public static Event createCleanUpEvent(String orgId, String eventType, OffsetDateTime cutOff) {
+    Event event = new Event();
+    event.setOrgId(orgId);
+    event.setEventSource(MeteringEventFactory.EVENT_SOURCE);
+    event.setEventType(CLEAN_UP_EVENT_TYPE_PREFIX + eventType);
```

Comment:
This is a controversial change. I did it because we're sending events by event type (for ex: instance hours and cores), therefore we need to only trigger the clean-up by event type as well. 

The approach I chose is to append the "cleanup-" text to the actual event type which I personally disliked but I didn't find any other cleaner way.

Reply:
About above, I've provided a new commit to refactor the above solution by adding a new optional "action" field: https://github.com/RedHatInsights/rhsm-subscriptions/pull/2456/commits/5375243d55de46bcbfbcf1ba42045c619abe0ea7

Let me know what you think about it. If you prefer going for the appending solution, we can drop the commit altogether.